Description: Move runtime documentation python modules into src/sage
 This is a temporary workaround; upstream is working on a proper fix.
 When you remove this patch, also be sure to remove the hacky entries in
 sagemath-common.install
Author: Ximin Luo <infinity0@debian.org>
Bug: https://trac.sagemath.org/ticket/21732
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/src/sage/misc/sagedoc.py
+++ b/sage/src/sage/misc/sagedoc.py
@@ -493,10 +493,15 @@
     """
     if embedded:
         return s
-    oldpath = sys.path
-    sys.path = [os.path.join(SAGE_DOC_SRC, 'common')] + oldpath
-    from conf import extlinks
-    sys.path = oldpath
+    extlinks = {
+        'python': ('https://docs.python.org/release/'+sys.version.split(' ')[0]+'/%s', ''),
+        'trac': ('https://trac.sagemath.org/%s', 'trac ticket #'),
+        'wikipedia': ('https://en.wikipedia.org/wiki/%s', 'Wikipedia article '),
+        'arxiv': ('http://arxiv.org/abs/%s', 'Arxiv '),
+        'oeis': ('https://oeis.org/%s', 'OEIS sequence '),
+        'doi': ('https://dx.doi.org/%s', 'doi:'),
+        'mathscinet': ('http://www.ams.org/mathscinet-getitem?mr=%s', 'MathSciNet ')
+        }
     for key in extlinks:
         while True:
             m = re.search(':%s:`([^`]*)`' % key, s)
--- a/sage/src/sage/misc/sphinxify.py
+++ b/sage/src/sage/misc/sphinxify.py
@@ -22,7 +22,7 @@
 import re
 import shutil
 from tempfile import mkdtemp
-from sage.env import SAGE_DOC_SRC
+from sage.env import SAGE_DOC_SRC, SAGE_LIB
 from sphinx.application import Sphinx
 
 
@@ -79,6 +79,8 @@
     # Sphinx constructor: Sphinx(srcdir, confdir, outdir, doctreedir,
     # buildername, confoverrides, status, warning, freshenv).
     confdir = os.path.join(SAGE_DOC_SRC, 'en', 'introspect')
+    if not os.path.exists(confdir):
+        confdir = os.path.join(SAGE_LIB, 'sage', 'misc', 'docs', 'introspect')
 
     open(os.path.join(srcdir, 'docutils.conf'), 'w').write(r"""
 [parsers]
--- a/sage/src/doc/en/introspect/conf.py
+++ b/sage/src/doc/en/introspect/conf.py
@@ -3,8 +3,11 @@
 # See sagenb.notebook.cell.Cell.set_introspect_html() for details.
 
 import sys, os
-sys.path.append(os.environ['SAGE_DOC_SRC'])
-from common.conf import *
+if 'SAGE_DOC_SRC' in os.environ: sys.path.append(os.environ['SAGE_DOC_SRC'])
+try:
+    from common.conf import *
+except ImportError:
+    from sage.misc.docs.common.conf import *
 
 extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo',
               'sphinx.ext.extlinks']
--- a/sage/src/doc/common/conf.py
+++ b/sage/src/doc/common/conf.py
@@ -1,11 +1,15 @@
 import sys, os, sphinx
-from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, THEBE_DIR
+from sage.env import SAGE_DOC_SRC, SAGE_DOC, SAGE_SRC, SAGE_LIB, THEBE_DIR
 from datetime import date
 from six import iteritems
 
 # If your extensions are in another directory, add it here.
 sys.path.append(os.path.join(SAGE_SRC, "sage_setup", "docbuild", "ext"))
 
+# If we can't find the doc sources, try in SAGE_LIB
+if not os.path.exists(SAGE_DOC_SRC):
+    SAGE_DOC_SRC = os.path.join(SAGE_LIB, 'sage', 'misc', 'docs')
+
 # General configuration
 # ---------------------
 
